services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -209,48 +233,53 @@ async def _start_dynamic_service(
     user_id: UserID,
     project_uuid: ProjectID,
     node_uuid: NodeID,
-):
+) -> None:
     if not _is_node_dynamic(service_key):
         return
-    project_running_nodes = await director_v2_api.list_dynamic_services(
-        request.app, user_id, f"{project_uuid}"
-    )
 
+    # this is a dynamic node, let's gather its resources and start it
+
+    lock_key = _get_service_start_lock_key(user_id, project_uuid)
+    client_sdk = get_redis_lock_manager_client_sdk(request.app)
     project_settings = get_settings(request.app).WEBSERVER_PROJECTS
-    assert project_settings  # nosec
-    if project_settings.PROJECTS_MAX_NUM_RUNNING_DYNAMIC_NODES > 0 and (
-        len(project_running_nodes)
-        >= project_settings.PROJECTS_MAX_NUM_RUNNING_DYNAMIC_NODES
+
```

Comment:
MINOR: this is perfectly correct but ``get_plugin_settings`` is the standard way

In this case the problem is that we still did not create this standard function. Do you mind to add  ``get_plugin_settings`` in ``project_settings.py`` as

```python
# projects_settings.py


def get_plugin_settings(app: web.Application) -> ProjectsSettings:
    settings = app[APP_SETTINGS_KEY].WEBSERVER_PROJECTS
    assert settings, "setup_settings not called?"  # nosec
    assert isinstance(settings, ProjectsSettings)  # nosec
    return settings
```

and import it here. thx

Reply:
Replaced with suggested pattern